From cfc37b970faa05c08b0b8c80f3a8e550769a3457 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Mon, 1 Nov 2021 18:10:35 +0100
Subject: [PATCH 03/21] xen-hvm: handle CONFIG_STUBDOM

Adapted from https://github.com/QubesOS/qubes-vmm-xen-stubdom-linux/blob/2a814bd1edaf549ef9252eb6747aa6137abf9831/qemu/patches/qemu-xen-hvm.patch

Original authors: HW42

Contributors: alcreator, fepitre, DemiMarie
---
 hw/xen/xen-hvm-common.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hw/xen/xen-hvm-common.c b/hw/xen/xen-hvm-common.c
index 565dc39c8f..f2f3c200c8 100644
--- a/hw/xen/xen-hvm-common.c
+++ b/hw/xen/xen-hvm-common.c
@@ -869,7 +869,15 @@ void xen_register_ioreq(XenIOState *state, unsigned int max_cpus,
 
     xen_bus_init();
 
+#ifndef CONFIG_STUBDOM
     xen_be_init();
+#else
+    xenstore = qemu_xen_xs_open();
+    if (!xenstore) {
+        error_report("can't connect to xenstored");
+        goto err;
+    }
+#endif
 
     return;
 
-- 
2.41.0

